### PR TITLE
INTDEV-541 Add support for labels in cli

### PR DESCRIPTION
**Pull request by samutoljamo-pareto**
Added support for the following commands:
cyberismo create label <cardId>
cyberismo remove label <cardId> [label name]
cyberismo show labels

### CODE REVIEW

File: tools/cli/src/index.ts [hunk 1]
```diff
@@ -219,6 +219,20 @@ create
     handleResponse(result);
   });
 
+create
+  .command('label')
+  .description('Create a label')
```
Comment (Fuzzbender) [inline tools/cli/src/index.ts:224]: Add descriptions for command parameters (just like you did with 'remove')

File: tools/cli/src/index.ts [hunk 2]
```diff
@@ -535,6 +549,20 @@ remove
...
 
+remove
+  .command('label')
+  .argument('<cardKey>', 'Source card key of the link')
+  .argument('[label]', 'Label being removed')
+  .option('-p, --project-path [path]', `${pathGuideline}`)
+  .action(async (cardKey: string, label: string, options: CardsOptions) => {
+    const result = await commandHandler.command(
+      Cmd.remove,
+      ['label', cardKey, label],
```
Comment (Fuzzbender) [inline tools/cli/src/index.ts:560]: Add 'label' to `RemovableResourceTypes`

File: tools/cli/src/index.ts [hunk 3]
```diff
@@ -642,7 +670,7 @@ program
   .description('Shows resource types in a project')
   .argument(
     '<type>',
-    'resource types: attachments, card, cards, cardType, cardTypes, linkType, linkTypes, project, reports, template, templates, workflow, workflows',
+    'resource types: attachments, card, cards, cardType, cardTypes, labels, linkType, linkTypes, project, reports, template, templates, workflow, workflows',
```
Comment (Fuzzbender) [inline tools/cli/src/index.ts:673]: Is there any point to show details of one type? Probably not if it is just one string.

File: tools/data-handler/src/create.ts [hunk 4]
```diff
@@ -408,6 +408,23 @@ export class Create extends EventEmitter {
     await this.createResource('fieldType', content);
   }
 
+  public async createLabel(cardKey: string, label: string) {
```
Comment (Fuzzbender) [inline tools/data-handler/src/create.ts:411]: 'label' should follow generic naming rules.
At least not empty string, must be at least N characters long and no longer than M characters. 
Should not contain extra characters.
There's at least `isValidResourceName()` function that could be used. It was intended for resource identifiers, but labels could follow similar rules.

File: tools/data-handler/src/show.ts [hunk 5]
```diff
@@ -269,6 +269,28 @@ export class Show {
...
+      }
+      // Recursively collect labels from subcards, if they exist
+      if (card.children) {
+        labels.push(...this.collectLabels(card.children));
+      }
+      return labels;
+    }, []);
+  };
+
+  public async showLabels(): Promise<string[]> {
```
Comment (Fuzzbender) [inline tools/data-handler/src/show.ts:286]: Add comment block to public API.

File: tools/data-handler/test/command-handler-create.test.ts [hunk 6]
```diff
@@ -392,6 +392,23 @@ describe('create command', () => {
...
+  it('create a label (success)', async () => {
+    const result = await commandHandler.command(
+      Cmd.create,
+      ['label', 'decision_6', 'test'],
+      options,
+    );
+    expect(result.statusCode).to.equal(200);
+  });
+
+  it('try create a label - label exists', async () => {
```
Comment (Fuzzbender) [inline tools/data-handler/test/command-handler-create.test.ts:404]: Add tests for invalid names (empty strings + if you add any other limitations; length, invalid characters).

**Review by Fuzzbender — APPROVED** [review]
Looks good. Some minor comments.

As labels can be added to template cards; then that template can be a in a module for another project. Thus, another project can receive labels from modules. 
Is this ok, or not? 
If yes - then labels should be collected also from template cards.
If not - then disallow setting a label to a template card.

**Comment by Fuzzbender** [comment]
> 
>If yes - then labels should be collected also from template cards.
Discussed with @henryhaverinen. Labels should be possible in template cards. Thus, ensure that the labels are also collected (in show) for templates. Perhaps add a test where unit test adds a label to a template card as well.

File: tools/cli/src/index.ts (hunk 3 above)
Comment (samutoljamo-pareto) [reply tools/cli/src/index.ts:673]: Maybe if you wanted a list of all cards with a specific label.

File: tools/cli/src/index.ts (hunk 1 above)
Comment (samutoljamo-pareto) [reply tools/cli/src/index.ts:224]: done

File: tools/data-handler/test/command-handler-create.test.ts (hunk 6 above)
Comment (samutoljamo-pareto) [reply tools/data-handler/test/command-handler-create.test.ts:404]: added those tests

**Comment by samutoljamo-pareto** [comment]
@Fuzzbender I noticed that the import from CSV, functionality overwrites the template labels, if they are defined. If they are not in the csv, labels are used from the template. I think I'll change it so that it also simply adds labels so it will have both the labels defined in the templates and in the csv
